### PR TITLE
DTLS 1.2: Reset state when sending HelloVerifyRequest

### PR DESCRIPTION
**Pull request by julek-wolfssl**

### CODE REVIEW

**Review by rizlik — CHANGES_REQUESTED** [review]
can you add a test?

File: src/dtls.c [hunk 1]
```diff
@@ -24,6 +24,52 @@
...
+    w64Zero(&ssl->dtls13Epochs[0].nextSeqNumber);
+    w64Zero(&ssl->dtls13Epochs[0].nextPeerSeqNumber);
+    XMEMSET(ssl->dtls13Epochs[0].window, 0,
+            sizeof(ssl->dtls13Epochs[0].window));
+    Dtls13FreeFsmResources(ssl);
+#endif
+    ssl->keys.dtls_expected_peer_handshake_number = 0;
+    ssl->keys.dtls_handshake_number = 0;
+
+    ssl->msgsReceived.got_client_hello = 0;
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply src/dtls.c:60]: Should only be once.

File: src/internal.c [hunk 2]
```diff
@@ -35041,6 +35042,8 @@ static int DefTicketEncCb(WOLFSSL* ssl, byte key_name[WOLFSSL_TICKET_NAME_SZ],
             int   inputSz = DTLS_HANDSHAKE_HEADER_SZ + length; /* build msg adds rec hdr */
             int   recordHeaderSz = DTLS_RECORD_HEADER_SZ;
 
+            ssl->msgsReceived.got_client_hello = 0;
```
[the remark this answers is not in the document]
Comment (julek-wolfssl) [reply src/internal.c:35045]: I was reading the function wrong. I thought that when `IsEncryptionOn(ssl, 1)` we are returning `sendSz` always. No need for this.

**Comment by julek-wolfssl** [comment]
Added testing (expanded `test_wolfSSL_dtls_stateless`).

**Review by rizlik — APPROVED** [review]

**Review by haydenroche5 — APPROVED** [review]

**Pull request merged**